src/mito2/src/cache.rs

```diff
@@ -55,6 +55,193 @@ const FILE_TYPE: &str = "file";
 /// Metrics type key for selector result cache.
 const SELECTOR_RESULT_TYPE: &str = "selector_result";
 
+/// Cache strategies that may only enable a subset of caches.
+#[derive(Clone)]
+pub enum CacheStrategy {
+    /// Strategy for normal operations.
+    /// Doesn't disable any cache.
+    Normal(CacheManagerRef),
+    /// Strategy for compaction.
+    /// Disables some caches that are not needed during compaction.
+    /// Enables the write cache so that the compaction can read files cached
+    /// in the write cache and write the compacted files back to the write cache.
+    Compaction(CacheManagerRef),
+    /// No cache.
+    Disabled,
+}
```

Comment:
naming strategy with target use scenarios seems misleading. Can we put how it actually works in the name? Like `EnableAll`, `MetaOnly`, `Disabled` (feel free to adjust these names)

Reply:
`EnableAll` is great. But the compaction needs a special cache strategy so I prefer to keep using `Compaction` as the strategy name.